eng/generate-compiler-code.ps1

```diff
@@ -42,6 +42,7 @@ function Run-LanguageCore($language, $languageSuffix, $languageDir, $syntaxProje
 function Test-GeneratedContent($generatedDir, $scratchDir) {
   $algo = "MD5"
   foreach ($fileName in (Get-ChildItem $scratchDir)) { 
+    $fileName = $fileName.Name
```

Comment:
Why was this change needed?

Reply:
On my local powershell installation (powershell core 7), Get-ChildItem is returning full file objects. So when you do Join-Path below you actually get:

C:\r\roslyn\src\generatedfoobar\C:\r\roslyn\src\generatedfoobar\Foo.cs

Reply:
I can definitely revert this if you want.

Reply:
I tweaked this so that it selects `Name` explicitly.